File: app/src/main/java/com/example/lance/simplebox/Adapter/InforAdapter.java
package com.example.lance.simplebox.Adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

/**
 * Created by Lance on 2017/12/16.
 */

public class InforAdapter extends FragmentPagerAdapter{

    private List<String> tablist;
    private List<Fragment> fragmentList;

    public InforAdapter(FragmentManager fm,List<String> tablist
            ,List<Fragment> fragmentList) {
        super(fm);
        this.tablist = tablist;
        this.fragmentList = fragmentList;
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return tablist.get(position % tablist.size());
    }
}

File: app/src/main/java/com/example/lance/simplebox/Adapter/InforAdapter.kt
package com.example.lance.simplebox.Adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * Created by Lance
 * on 2017/12/16.
 */

class InforAdapter(fm: FragmentManager, private val tablist: List<String>, private val fragmentList: List<Fragment>) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return fragmentList[position]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return tablist[position % tablist.size]
    }
}
